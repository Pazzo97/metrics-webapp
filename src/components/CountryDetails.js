import { React, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getDataFromApi } from '../redux/metric-redux';

const CountryDetails = () => {
  const listing = useSelector((state) => state.metricsReducer);
  const { id } = useParams();

  const dispatch = useDispatch();

  const details = listing.filter((country) => country.id === Number(id));

  useEffect(() => {
    if (listing.length === 0) {
      getDataFromApi(dispatch);
    }
  }, [dispatch]);

  return (
    <div className="country">
      {details.map((crt) => (
        <div key={0}>
          <div className="countryHeading">
            <div className="countryHeadingImage">
              <img src={crt.flags} alt="Africa map" />
            </div>
            <div className="countryHeadingTitle">
              <div>
                <h1>{crt.shortName}</h1>
                <span>{crt.fullName}</span>
              </div>
            </div>
          </div>
          <div className="countrylistTitle">
            <h3>COUNTRY DETAILS</h3>
          </div>
          <table key={1}>
            <tbody>
              <tr key={3}>
                <td className="countryDetails">
                  <span>Currency:</span>
                  <span>{crt.currencyName}</span>
                </td>
              </tr>
              <tr key={4}>
                <td className="countryDetails">
                  <span>Currency Symbol:</span>
                  <span>{crt.currencySymbol}</span>
                </td>
              </tr>
              <tr key={5}>
                <td className="countryDetails">
                  <span>Capital:</span>
                  <span>{crt.capital}</span>
                </td>
              </tr>
              <tr key={6}>
                <td className="countryDetails">
                  <span>Sub Region:</span>
                  <span>{crt.subregion}</span>
                </td>
              </tr>
              <tr key={7}>
                <td className="countryDetails">
                  <span>Language:</span>
                  <span>{crt.languages}</span>
                </td>
              </tr>
              <tr key={8}>
                <td className="countryDetails">
                  <span>Population:</span>
                  <span>{crt.population}</span>
                </td>
              </tr>
              <tr key={9}>
                <td className="countryDetails">
                  <span>Fifa Symbol:</span>
                  <span>{crt.fifa}</span>
                </td>
              </tr>
              <tr key={10}>
                <td className="countryDetails">
                  <span>Car Sign:</span>
                  <span>{crt.carSigns}</span>
                </td>
              </tr>
              <tr key={11}>
                <td className="countryDetails">
                  <span>Car Side:</span>
                  <span>{crt.carSide}</span>
                </td>
              </tr>
              <tr key={12}>
                <td className="countryDetails">
                  <span>Timezone:</span>
                  <span>{crt.timezones}</span>
                </td>
              </tr>
              <tr key={13}>
                <td className="countryDetails">
                  <span>Continents:</span>
                  <span>{crt.continents}</span>
                </td>
              </tr>
              <tr key={14}>
                <td className="countryDetails">
                  <span>Start of week:</span>
                  <span>{crt.startOfWeek}</span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      ))}
    </div>
  );
};

export default CountryDetails;
