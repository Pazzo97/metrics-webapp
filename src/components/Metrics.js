import { React, useEffect, useState } from 'react';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowAltCircleRight } from '@fortawesome/free-solid-svg-icons';
import { getDataFromApi, countryFilter } from '../redux/metric-redux';
import africa from '../images/africa.png';

const Metrics = () => {
  const listing = useSelector((state) => state.metricsReducer, shallowEqual);

  const [countryTitle, setcountryTitle] = useState('');

  const dispatch = useDispatch();

  const handleInputTitle = (e) => {
    e.preventDefault();
    setcountryTitle(e.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(countryFilter(countryTitle));
  };

  useEffect(() => {
    if (listing.length === 0) {
      getDataFromApi(dispatch);
    }
  }, [dispatch]);

  return (
    <div className="country">
      <div className="countryHeading">
        <div className="countryHeadingImage">
          <img src={africa} alt="Africa map" />
        </div>
        <div className="countryHeadingTitle">
          <div>
            <h1>AFRICA</h1>
            <br />
            <span>
              A continent that spans the hemispheres
              {' '}
              <br />
              Home to some of the worldâ€™s oldest civilisations
            </span>
          </div>
        </div>
      </div>
      <div className="countrylistTitle">
        <h3>LIST OF COUNTRIES</h3>
        <form className="book-form" onSubmit={handleSubmit}>
          <input
            type="text"
            className="inputcountryTitle searchBox"
            placeholder="Search by country"
            onChange={handleInputTitle}
            value={countryTitle}
          />
          <button className="submit-btn seachButton" type="submit">
            Search
          </button>
        </form>
        <button
          className="submit-btn clearButton"
          type="button"
          onClick={() => getDataFromApi(dispatch)}
        >
          ClearSearch
        </button>
      </div>
      <ul>
        {listing.map((country) => (
          <li
            className="countrylistItem"
            style={{
              backgroundImage: `url(${country.flags})`,
            }}
            key={country.id}
          >
            <NavLink
              className="active-link"
              to={`/countryDetails/${country.id}`}
            >
              <div>
                <FontAwesomeIcon icon={faArrowAltCircleRight} />
              </div>
              <div>
                <h2>{country.shortName}</h2>
                <span>{country.capital}</span>
              </div>
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Metrics;
