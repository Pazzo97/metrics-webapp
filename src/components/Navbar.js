import React from 'react';
import { useNavigate } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faChevronLeft,
  faMicrophone,
  faCog,
} from '@fortawesome/free-solid-svg-icons';

const Navbar = () => {
  const navigate = useNavigate();
  return (
    <header>
      <nav>
        <div
          style={{ cursor: 'pointer' }}
          aria-hidden
          onClick={() => navigate(-1)}
        >
          <span className="leftNav">
            <FontAwesomeIcon icon={faChevronLeft} />
            Africa
          </span>
        </div>
        <div>
          <span className="middleNav">Get to know about Africa</span>
        </div>
        <div>
          <span className="rightNav">
            <FontAwesomeIcon icon={faMicrophone} />
            <FontAwesomeIcon icon={faCog} />
          </span>
        </div>
      </nav>
    </header>
  );
};

export default Navbar;
