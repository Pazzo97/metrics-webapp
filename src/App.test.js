import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Metrics from './components/Metrics';
import CountryDetails from './components/CountryDetails';
import Navbar from './components/Navbar';
import store from './redux/configureStore';
import '@testing-library/jest-dom';

const MetricsProvider = () => (
  <Provider store={store}>
    <Metrics />
  </Provider>
);

const CountryDetailsProvider = () => (
  <Provider store={store}>
    <CountryDetails />
  </Provider>
);

const Navigation = () => (
  <Router>
    <Navbar />
  </Router>
);

describe('component test', () => {
  it('Metrics renders correctly', () => {
    const tree = renderer.create(<MetricsProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CountryDetails renders correctly', () => {
    const tree = renderer.create(<CountryDetailsProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Metrics screen renders correctly', async () => {
    render(<MetricsProvider />);
    expect(screen.getByText('AFRICA')).toBeInTheDocument();
  });

  it('Navbar renders correctly', async () => {
    render(<Navigation />);
    expect(screen.getByText('Get to know about Africa')).toBeInTheDocument();
  });

  test('NavLink to match snapshot', () => {
    const component = renderer.create(
      <Router>
        <NavLink to="/countryDetails/1" />
      </Router>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
