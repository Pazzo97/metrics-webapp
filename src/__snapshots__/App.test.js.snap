// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test CountryDetails renders correctly 1`] = `
<div
  className="country"
/>
`;

exports[`component test Metrics renders correctly 1`] = `
<div
  className="country"
>
  <div
    className="countryHeading"
  >
    <div
      className="countryHeadingImage"
    >
      <img
        alt="Africa map"
        src="africa.png"
      />
    </div>
    <div
      className="countryHeadingTitle"
    >
      <div>
        <h1>
          AFRICA
        </h1>
        <br />
        <span>
          A continent that spans the hemispheres
           
          <br />
          Home to some of the worldâ€™s oldest civilisations
        </span>
      </div>
    </div>
  </div>
  <div
    className="countrylistTitle"
  >
    <h3>
      LIST OF COUNTRIES
    </h3>
    <form
      className="book-form"
      onSubmit={[Function]}
    >
      <input
        className="inputcountryTitle searchBox"
        onChange={[Function]}
        placeholder="Search by country"
        type="text"
        value=""
      />
      <button
        className="submit-btn seachButton"
        type="submit"
      >
        Search
      </button>
    </form>
    <button
      className="submit-btn clearButton"
      onClick={[Function]}
      type="button"
    >
      ClearSearch
    </button>
  </div>
  <ul />
</div>
`;

exports[`component test NavLink to match snapshot 1`] = `
<a
  className=""
  href="/countryDetails/1"
  onClick={[Function]}
/>
`;
